src/main/java/uk/gov/hmcts/reform/unspec/service/docmosis/DocumentGeneratorService.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.reform.unspec.service.docmosis;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.unspec.config.DocmosisConfiguration;
+import uk.gov.hmcts.reform.unspec.model.docmosis.DocmosisData;
+import uk.gov.hmcts.reform.unspec.model.docmosis.DocmosisDocument;
+import uk.gov.hmcts.reform.unspec.model.docmosis.DocmosisRequest;
+
+import java.util.Map;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
```

Comment:
No need to explicitly add `@Autowired` on single constructor in a component.  
```suggestion
@RequiredArgsConstructor
```

Reply:
done